templates/sshd_config.erb

```diff
@@ -12,9 +12,13 @@
 # OpenSSH is to specify options with their default value where
 # possible, but leave them commented.  Uncommented options change a
 # default value.
-
+<% if port = scope.lookupvar("ssh::sshd_config_port") and port.any? %>
+<% port.each do |p| -%>
+<%= "Port #{p}" %>
+<% end -%>
+<%else -%>
 #Port 22
-Port <%= @sshd_config_port %>
```

Comment:
You could use `.class` and if it is string do variable interpolation and if an array then iteration.